### PR TITLE
Test published/unpublished cases

### LINKED ISSUE #3573: Add filter to only show published cases on /legal/
Salient is working on a new version of SMUR, the internal MUR management software. Users of this software need to be able to publish/unpublish a case (MUR/ADR/AF). Currently, when a user unpublishes a MUR, a hard delete is done in postgres. However, in order for the celery task to be able to detect an unpublished case and reload elasticsearch to reflect the change,  we need to add an `is_published` flag to the `fecmur.case` table, and only show published cases. Salient will update the `is_published` and `pg_update` date.

**Completion criteria**
**DB changes**
- [x] Add a new migration file
- [x] Update [`"case"` table](https://github.com/fecgov/openFEC/blob/develop/data/migrations/V0021__fecmur_schema_objects.sql#L67-L74) to add `is_published` boolean column (Salient will update this data. Please **make sure** Salient updates the `pg_update` field when they publish/unpublish the case.) 
- [x] Update [`fecmur.cases_with_parsed_case_serial_numbers_vw`](https://github.com/fecgov/openFEC/blob/develop/data/migrations/V0105__add_adr_af.sql#L8-L16) to include the column so we can detect when cases are unpublished

**Python changes:**
- [x] Update [queries](https://github.com/fecgov/openFEC/blob/develop/webservices/legal_docs/current_cases.py#L15-L150) to only show published (celery task shouldn't need changes - it only looks at the `pg_update` field)
- [x] Double-check subtable behavior (including `af_case` table) for published/unpublished cases
- [x] Double-check elasticsearch behavior when we unpublish a case - does the reload remove it?
- [x] Add an automated test to make sure unpublishing/publishing a case will result in hiding/showing a case

### PR DESCRIPTION
**Pull request by pkfec**
## Summary (required)
A case (ADR, AF or MUR) is purged on  DB, by setting the `published_flg=false` and updating `pg_date` column in case table. This will initiate a celery task which deletes the case from elasticsearch service.

- Resolves #3573 

_Include a summary of proposed changes._

  1. Modify the case queries to fetch the new boolean column published_flg
  2. Update the `load_cases` script to check if `published_flg=false` set  on a case and delete it from ES service.
  3. Update `test_current_cases.py` test to return the published_flg for a given case 
 
## How to test the changes locally
  1.Follow the instructions and set up your workspace with the following services:
       (https://github.com/fecgov/openFEC/wiki/Set-up-Redis,--Celery-on-local-and-test-the-downloads#testing-queued-tasks-locally)
  2.start your local elasticsearch service
  3.get the API s3 bucket env variables from `dev` space (`cf env api`)
  4.export the access, secretaccesskey, bucket, region env variables in  `celery worker` and `api` 
    terminals
5.export sqla_conn to point to your local db
6. load a case(ex: mur 7278) onto your local elasticsearch service 
7. verified that 7278 existing on  local ES service
8. on cfdm_test db, updated the two columns `pg_date` and `processed_flg=false` (which means that the case is purged in the DB) this will initiate the celery task.
9. refresh the local ES service and make sure the mur 7278 is deleted.
10. try to reload the deleted mur 7278 again. the load_cases script will not LOAD this MUR since its already deleted. 
11. Run the steps 6-10 to test ADR's or AF's

## Impacted areas of the application
List general components of the application that this PR will affect:
- Legal Resources/ Enforcements
 https://www.fec.gov/data/legal/search/enforcement/?case_no=

### CODE REVIEW

**Comment by pkfec** [comment]
@lbeaufort Please review.

File: webservices/tasks/legal_docs.py [hunk 1]
```diff
@@ -98,10 +98,15 @@ def refresh_cases(conn):
     rs = conn.execute(RECENTLY_MODIFIED_CASES)
     if rs.returns_rows:
         load_count = 0
+        deleted_case_count = 0
         for row in rs:
             logger.info("%s %s found modified at %s", row["case_type"], row["case_no"], row["pg_date"])
             load_cases(row["case_type"], row["case_no"])
-            load_count += 1
-        logger.info("Total of %d case(s) loaded", load_count)
+            if row["published_flg"]:
```
Comment (lbeaufort) [inline webservices/tasks/legal_docs.py]: @pkfec should we put the total logging statements outside the `for` loop like we did on line 105 before?

https://github.com/fecgov/openFEC/blob/7c7454ad2a4fe06f73d7eefabdca9c2f6b21d051/webservices/tasks/legal_docs.py#L102-L110

Comment (pkfec) [reply webservices/tasks/legal_docs.py]: @lbeaufort 

I tried to load AF 7283, and unpublish MUR's 7281, 7297 on my local `cfdm_test` db. Below are the log statements which show the count of case(s) that are loaded and unpublished.  

[2019-04-26 16:35:00,005: INFO/ForkPoolWorker-4] MUR 7281 found modified at 2019-04-27 15:50:00
[2019-04-26 16:35:00,005: INFO/ForkPoolWorker-4] Loading MUR(s)
[2019-04-26 16:35:00,069: INFO/ForkPoolWorker-4] Found an unpublished case - deleting from ES
[2019-04-26 16:35:00,074: INFO/ForkPoolWorker-4] Sucessfully deleted MUR 7281 from ES
**[2019-04-26 16:35:00,075: INFO/ForkPoolWorker-4] Total of 1 case(s) unpublished...**
[2019-04-26 16:35:00,075: INFO/ForkPoolWorker-4] AF 7283 found modified at 2019-04-27 15:50:00
[2019-04-26 16:35:00,075: INFO/ForkPoolWorker-4] Loading AF(s)
[2019-04-26 16:35:00,138: INFO/ForkPoolWorker-4] Loading AF: 7283
[2019-04-26 16:35:00,150: INFO/ForkPoolWorker-4] 1 AF(s) loaded
**[2019-04-26 16:35:00,151: INFO/ForkPoolWorker-4] Total of 1 case(s) loaded...**
[2019-04-26 16:35:00,151: INFO/ForkPoolWorker-4] MUR 7297 found modified at 2019-04-27 15:50:00
[2019-04-26 16:35:00,152: INFO/ForkPoolWorker-4] Loading MUR(s)
[2019-04-26 16:35:00,235: INFO/ForkPoolWorker-4] Found an unpublished case - deleting from ES
[2019-04-26 16:35:00,241: INFO/ForkPoolWorker-4] Sucessfully deleted MUR 7297 from ES
**[2019-04-26 16:35:00,241: INFO/ForkPoolWorker-4] Total of 2 case(s) unpublished...**

**Comment by lbeaufort** [comment]
Great work, Priya! I think the totals make sense as a final total (outside the loop) rather than a running total (inside the loop) but I defer to you.

**Pull request merged**